types/module/module.go

```diff
@@ -344,10 +344,14 @@ func (m *Manager) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) map[string
 // MigrationHandler is the migration function that each module registers.
 type MigrationHandler func(sdk.Context) error
 
-// VersionMap is a map of moduleName -> version, where version denotes the
-// version from which we should perform the migration for each module.
+// VersionMap is a map of moduleName -> version
 type VersionMap map[string]uint64
 
+type MigrationOrder struct {
```

Comment:
I have a slight preference for simply adding a `SetOrderInPlaceMigration([]string)`, for these reasons:
- it's not API-breaking
- it's similar to the current way of doing things `SetOrderBeginBlock`, `SetOrderGenesis`...
- it avoids introducing a new notion of "priority", which comes with its own hidden-behavior questions (e.g. "what happens when 2 modules have same priority? alphabetical?", "why is in-place migration special and other SetOrder* don't have priority?"...). These questions all need documentation.

Reply:
+1 preference towards having a slice of module names denoting order, seems the path of least resistance